src/include/main/db_main.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <network/terrier_server.h>
+#include <memory>
+#include <unordered_map>
+#include <utility>
+#include "catalog/catalog.h"
+#include "common/action_context.h"
+#include "common/stat_registry.h"
+#include "settings/settings_param.h"
+#include "transaction/transaction_manager.h"
+
+namespace terrier {
+
+namespace settings {
+class SettingsManager;
+class SettingsTests;
+}  // namespace settings
+
+/**
+ * The DBMain Class holds all the static functions, singleton pointers, etc.
+ * It has the full knowledge of the whole database systems.
+ * *Only the settings manager should be able to access the DBMain object.*
+ */
+class DBMain {
```

Comment:
If one DBMain only correspond to one settings manager, it might be a good idea to merge these two classes into one.

Reply:
I think it will make the class too big. I would insist that DBMain and Settings Manager be two classes.